src/commands/dev.rs

```diff
@@ -1,22 +1,150 @@
+use std::path::PathBuf;
+
+use axoproject::WorkspaceSearch;
 use camino::Utf8PathBuf;
 use clap::Parser;
+use mdbook::MDBook;
+use notify::{event::ModifyKind, EventKind, Watcher};
 
-use crate::commands::{Build, Serve};
-use oranda::errors::*;
+use crate::{
+    commands::{Build, Serve},
+    message::{Message, MessageType},
+};
+use oranda::{config::Config, errors::*};
 
 #[derive(Clone, Debug, Parser)]
 pub struct Dev {
+    /// The port for the file server to be launched on
     #[arg(long)]
     port: Option<u16>,
+    /// The project root we want to build from
     #[arg(long)]
     project_root: Option<Utf8PathBuf>,
+    /// Custom path to an oranda configuration file
     #[arg(long)]
     config_path: Option<Utf8PathBuf>,
+    /// Skip the first build before starting to watch for changes
+    #[arg(long)]
+    no_first_build: bool,
+    /// List of extra paths to watch
+    #[arg(short, long)]
+    include_paths: Option<Vec<Utf8PathBuf>>,
 }
 
 impl Dev {
     pub fn run(self) -> Result<()> {
-        Build::new(self.project_root.clone(), self.config_path.clone()).run()?;
-        Serve::new(self.port).run()
+        Message::new(
+            MessageType::Info,
+            "Starting dev, looking for paths to watch...",
+        )
+        .print();
+        tracing::info!("Starting dev, looking for paths to watch...");
+
+        let config = Config::build(
+            &self
+                .config_path
+                .clone()
+                .unwrap_or(Utf8PathBuf::from("./oranda.json")),
+        )?;
+        let mut paths_to_watch = vec![];
+        // Watch for the readme file
+        paths_to_watch.push(config.readme_path);
+        // Watch for the oranda config file
+        paths_to_watch.push(
+            self.config_path
+                .clone()
+                .unwrap_or(Utf8PathBuf::from("./oranda.json"))
+                .into(),
+        );
+
+        // Watch for any user-provided paths
+        if self.include_paths.is_some() {
+            let mut include_paths: Vec<String> = self
+                .include_paths
+                .unwrap()
+                .iter()
+                .map(|p| p.to_string())
+                .collect();
+            paths_to_watch.append(&mut include_paths);
+        }
+
+        // Watch for additional pages, if we have any
+        if config.additional_pages.is_some() {
+            let mut additional_pages: Vec<String> =
+                config.additional_pages.unwrap().values().cloned().collect();
+            paths_to_watch.append(&mut additional_pages);
+        }
+
+        // Watch for the mdbook directory, if we have it
+        if config.mdbook.is_some() {
+            let path = config.mdbook.unwrap().path;
+            let md = MDBook::load(&path).map_err(|e| OrandaError::MdBookLoad { path, inner: e })?;
+            paths_to_watch.push(md.source_dir().display().to_string());
+        }
```

Comment:
```suggestion
        if let Some(book_cfg) = &config.mdbook {
            let path = &book_cfg.path;
            let md = MDBook::load(&path).map_err(|e| OrandaError::MdBookLoad { path, inner: e })?;
            
            // watch book.toml and /src/
            paths_to_watch.push(md.root.join("book.toml").display().to_string());
            paths_to_watch.push(md.source_dir().display().to_string());
            
            // If we're not clobbering the theme, also watch the theme dir
            // (note that this may not exist on the fs, mdbook reports the path regardless)
            if !book_cfg.theme.unwrap_or(true) {
                paths_to_watch.push(md.theme_dir().display().to_string());
            }
        }
```

(although ideally watching book.toml would be non-recursive? does that matter?)

Reply:
afaik `notify` doesn't even consider the recursive setting on files, so it doesn't matter